aips/staking-contract-to-delegation-pool.md

```diff
@@ -0,0 +1,145 @@
+- `destroy_staking_contract` and take ownership of `StakingContract`'s contents
+- converts `commission_percentage` to a 2-decimals precision format used within `delegation_pool` module
+- stores `owner_cap` on the resource account of `signer_cap`
+- creates the `delegation_pool::DelegationPool` resource:
+    * initialize `active_shares` and `inactive_shares` shares pools
+    * initialize `observed_lockup_cycle`
+    * initialize `pending_withdrawals`
+    * set `stake_pool_signer_cap` to `signer_cap`
+    * set `operator_commission_percentage` to `commission_percentage`
+    * set `total_coins_inactive` to 0: `distribute_internal` call ensures there is no *inactive* stake to be accounted for
+- allocates *active* shares, equivalent to `principal`, to *staker* in `active_shares` pool
+- allocates *pending-inactive* shares to each shareholder of `distribution_pool` in `inactive_shares(0)` pool
+- stores `DelegationPool` resource on the resource account of `signer_cap` also storing `StakePool`
+- stores `DelegationPoolOwnership` capability on *staker*
+- sets delegated voter of *staker* to their previous voter on the staking contract without waiting a lockup cycle to end on the stake pool
+
+`delegation_pool::BeneficiaryForOperator` could not have been created + published because *operator*'s signer is inaccessible.
+If the *operator* already has a beneficiary set for their delegation pools then use that address, otherwise *operator*'s address will be used by default.
+
+## Reference Implementation
+
+There is a reference implementation at https://github.com/bwarelabs/aptos-core/tree/convert-staking-contract-to-delegation-pool
+
+## Testing (Optional)
+
+Design UTs to validate that:
+- there is no remaining state of the staking contract while the underlying stake pool remains untouched
+- the post-conversion state is consistent
+- the resulted delegation pool accepts delegations, staker is treated as a regular delegator and the commission is correctly distributed
+
+## Risks and Drawbacks
+
+The beneficiary of operator is set to a possibly different address than the one used on the staking contract without operator's approval. This address is either the one already configured on the delegation-pool module or the operator themselves, which is an address still owned by the operator.
+
+## Future Potential
+
+All existing staking contracts would be converted to delegation pools and create a homogenous validator landscape.
+
+## Timeline
+
+### Suggested implementation timeline
+
+### Suggested developer platform support timeline
+
+The conversion is a one-time process that could be executed through a wallet connected to the Aptos explorer or via CLI.
+
+### Suggested deployment timeline
+
+## Security Considerations
+
+If the conversion can be executed only by the staker, they might decide to not proceed with it, thereby preventing the operator from acquiring additional stake.
```

Comment:
How is this a security concern? Isn't it already true that the conversion can be executed only by the staker?

Reply:
that's true, it is not really a security issue, but rather a mention that only the staker can execute the conversion